dotCMS/src/main/java/com/dotcms/uuid/shorty/ShortyIdAPIImpl.java

```diff
@@ -87,7 +84,8 @@ public String randomShorty() {
   @Override
   public String shortify(final String shortStr) {
     try {
-      return shortStr.trim().replaceAll("-", "").substring(0, MINIMUM_SHORTY_ID_LENGTH);
+      return UtilMethods.isSet(shortStr)?
+              shortStr.trim().replaceAll("-", "").substring(0, MINIMUM_SHORTY_ID_LENGTH): shortStr;
```

Comment:
This will throw an IndexOutOfBounds if you pass in a 5 char shorty (for example). We should trim and return the lesser of the length of the shortStr vs. the MINIMUM_SHORTY_ID_LENGTH